### ISSUE TITLE
[Question / Feature Request] Alternative sources for slow downloads

### ISSUE THREAD

**Issue by Searinox** [zlatinb/muwire#75]
I noticed that it is possible to copy and search for files by hash. If a download is in progress and it is slow, will MuWire check for other users on the network that are sharing it and attempt to download pieces from them too?

Going into future upgrades to the download structure in order to allow for directory downloads this makes sense, because when you download a directory it doesn't have a hash and initially all the hashes of the files contained within are obtained from the one MuWire instance it's being downloaded from. If some files are particularly slow to download - either because they're big or because the host itself is slow, or even goes offline - is MuWire able to automatically search for alternate sources and start downloading from them as well?

**Comment by zlatinb** [comment]
Yes, one piece being downloaded from one user while another piece from another user is how it works.  Maybe it's not clear in the UI, but when you click on the download row in the downloads table, in the bottom of the tab in the "Download details" panel you will see "Known sources" and "Active sources" - the known sources are the total sources that MuWire thinks have the file, and active sources are the ones it currently is downloading from.
